core/src/test/scala/pink/cozydev/lucille/WildCardParserSuite.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 CozyDev
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pink.cozydev.lucille
+import Query._
+import WildCardOp._
+import cats.data.NonEmptyList
+
+class WildCardParserSuite extends munit.FunSuite {
+
+  // We already have a `Prefix` query
+  // do we make this a special case of the larger `Wildcard` query?
+  val parseQ = QueryParser.withDefaultOperatorOR.parse(_)
+
+  test("trailing single char") {
+    val r = parseQ("cat?")
+    assertEquals(r, Right(WildCard(NonEmptyList.of(Str("cat"), SingleChar))))
+  }
+
+  test("trailing many char") {
+    val r = parseQ("cat*")
+    assertEquals(r, Right(Prefix("cat")))
+  }
```

Comment:
Could we also have a test / demo of `?cat*`, since in that case it would not be a prefix query?

Reply:
Nice catch. Added a handful more in https://github.com/cozydev-pink/lucille/pull/154/commits/787d4dba1947102dde334b8e74439da6bc80f47d